### PR TITLE
Translation mode: Add Notification mode

### PR DESCRIPTION
**Pull request by rainbowhxch**
Hello, thank you very much for writing such a good tool.

I have been using i3-wm under Linux, but when I use crow, the main translation window that pops up often changes the layout of opened applications. So, I've always wanted crow to have a function to display translation results based on system notification. I added some code on the basis of the program and wrote a demo. I don't know if this can be useful or if you have plans to add it. Please feel free to choose whether to use it or not. I'm a new QT player, so I don't know if there's a better way.

### CODE REVIEW

File: src/mainwindow.cpp [hunk 1]
```diff
@@ -379,6 +379,12 @@ void MainWindow::parseTranslation()
         ui->translationLanguagesWidget->setAutoLanguage(m_translator->translationLanguage());
     else
         ui->translationLanguagesWidget->setAutoLanguage(QOnlineTranslator::Auto);
+
+    // If window mode is notification, send a notification including the translation result
+    const AppSettings settings;
+    if (this->isHidden() && settings.windowMode() == AppSettings::Notification) {
+        m_trayIcon->showMessage(tr("Translation Result"), ui->translationEdit->toPlainText(), QSystemTrayIcon::NoIcon, 3000);
+    }
```
Comment (Shatur95) [inline src/mainwindow.cpp:387]: In [Qt Coding style](https://wiki.qt.io/Qt_Coding_Style) curly braces used only when the body of a conditional statement contains more than one line:
```suggestion
    if (this->isHidden() && settings.windowMode() == AppSettings::Notification)
        m_trayIcon->showMessage(tr("Translation Result"), ui->translationEdit->toPlainText(), QSystemTrayIcon::NoIcon, 3000);
```

Also you specified `QSystemTrayIcon::NoIcon` and `3000`. Is there a reason to display a message without icon and only for a 3 sec?
By default function accepts `QSystemTrayIcon::Information` and `10000`.

File: src/mainwindow.cpp [hunk 2]
```diff
@@ -434,6 +441,10 @@ void MainWindow::showTranslationWindow()
             if (!ui->autoTranslateCheckBox->isChecked())
                 ui->sourceEdit->setRequestTranlationOnEdit(false);
         });
+    } else if (this->isHidden() && settings.windowMode() == AppSettings::Notification) {
+        // If window mode is notification, then dirictly return
+        m_translateSelectionHotkey->blockSignals(false);
+		return;
```
Comment (Shatur95) [inline src/mainwindow.cpp:447]: You accidentally used tabs here. Please use spaces :)
```suggestion
        return;
```

File: src/settings/settingsdialog.ui [hunk 3]
```diff
@@ -6,8 +6,8 @@
    <rect>
     <x>0</x>
     <y>0</y>
-    <width>800</width>
-    <height>552</height>
+    <width>837</width>
+    <height>586</height>
```
Comment (Shatur95) [inline src/settings/settingsdialog.ui:10]: Were these changes intentional?

**Review by Shatur95 — CHANGES_REQUESTED** [review]
Hello! thank you a lot for the PR. I like the idea.

**Comment by Shatur95** [comment]
Also you displayed message in void `MainWindow::parseTranslation()`. This is the right place, but I think that we should rename this function, because now it does not only parse the translation. I suggest renaming it to `MainWindow::displayTranslation`.

File: src/mainwindow.cpp (hunk 1 above)
Comment (rainbowhxch) [reply src/mainwindow.cpp:387]: I don't think the icon has much meaning for the translation results, so I removed the icon. Moreover, 10s is too long for quick view, and the translation results will block other applications during display, so I chose 3S temporarily. However, I think we can also add an option of "Notification display time" to the "Settings".

File: src/mainwindow.cpp (hunk 2 above)
Comment (rainbowhxch) [reply src/mainwindow.cpp:447]: I will correct it.

File: src/settings/settingsdialog.ui (hunk 3 above)
Comment (rainbowhxch) [reply src/settings/settingsdialog.ui:10]: I may have changed it by accident and will correct it

**Comment by rainbowhxch** [comment]
> Also you displayed message in void `MainWindow::parseTranslation()`. This is the right place, but I think that we should rename this function, because now it does not only parse the translation. I suggest renaming it to `MainWindow::displayTranslation`.

That's well be better

File: src/mainwindow.cpp (hunk 1 above)
Comment (Shatur95) [reply src/mainwindow.cpp:387]: Good point. For now I suggest to leave 3 seconds :)

Comment (Shatur95) [reply src/mainwindow.cpp:387]: In [Qt Coding style](https://wiki.qt.io/Qt_Coding_Style) curly braces used only when the body of a conditional statement contains more than one line.
```suggestion
    if (this->isHidden() && settings.windowMode() == AppSettings::Notification)
        m_trayIcon->showMessage(tr("Translation Result"), ui->translationEdit->toPlainText(), QSystemTrayIcon::NoIcon, 3000);
```

Comment (Shatur95) [reply src/mainwindow.cpp:387]: > Good point. For now I suggest to leave 3 seconds :)

I meant that such settings would be nice, but I suggest to implement it into another PR.

Comment (rainbowhxch) [reply src/mainwindow.cpp:387]: OK, I see. I will revise them and submit them again.

Comment (Shatur95) [reply src/mainwindow.cpp:387]: To be more clear: we will merge this PR first after your corrections and "Notification display time" feature can be added into a separate PR.

Comment (rainbowhxch) [reply src/mainwindow.cpp:387]: Ok, I got it.

**Comment by rainbowhxch** [comment]
Thank you for your comments. I have modified the code. Please check it.

**Comment by Shatur95** [comment]
Thanks! I only updated the translation source file.

**Comment by rainbowhxch** [comment]
No problem, thanks a lot.
